### PR TITLE
Stop-gap fixes for compiler warnings

### PR DESCRIPTION
**Pull request by alradmsft**
Related to #45. This is just to show the lines that I had to change to get a clean build on VS2015 compiler.

### CODE REVIEW

File: src/cborencoder.c [hunk 1]
```diff
@@ -290,7 +290,7 @@ static inline CborError encode_number_no_update(CborEncoder *encoder, uint64_t u
     if (ui < Value8Bit) {
         *bufstart += shiftedMajorType;
     } else {
-        unsigned more = 0;
+		uint8_t more = 0;
```
Comment (thiagomacieira) [inline src/cborencoder.c]: Please fix the whitespace here (TinyCBOR uses only spaces, no tab).

File: src/compilersupport_p.h [hunk 2]
```diff
@@ -210,7 +210,15 @@ static inline unsigned short encode_half(double val)
         /* underflow, make zero */
         return 0;
     }
+
+//
+// bugbug: working with tinycbor folks to address this properly. 
+// Issue https://github.com/01org/tinycbor/issues/47
+//
+#pragma warning(disable:4244)
```
Comment (thiagomacieira) [inline src/compilersupport_p.h]: I can't accept an MSVC pragma here, not without #ifndef _MSC_VER. Is there no other way of getting this? Maybe

  return (unsigned short)(sign | ((exp + 15) << 10) | mant);

**Comment by alradmsft** [comment]
Sorry I didn't realize that you were going to pull this actually. I thought you only were looking for the list of warnings so I hacked something up really quick. You got me embarrassed with spaces actually. :-)

I can work on the proper change, no problem. For the compilersupport_p.h, I used pragma to just get my build working so I can play with the lib. Doing a cast + comment as to why it is safe is OK. I did look at DirectX implementation of this function and uses all unsigned integers so it doesn't have this problem. But I personally don't care for such a major re-write.

**Comment by alradmsft** [comment]
Will start another PR.

**Pull request closed without merging**